test/tools/ossfuzz/proto_to_yul.cpp

```diff
@@ -0,0 +1,234 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/ossfuzz/proto_to_yul.h>
+#include <test/tools/ossfuzz/yul_proto.pb.h>
+
+#include <ostream>
+#include <sstream>
+
+namespace yul_fuzzer {
+
+	// Forward decls.
+	std::ostream &operator<<(std::ostream &os, const BinaryOp &x);
+
+	std::ostream &operator<<(std::ostream &os, const StatementSeq &x);
+
+	// Proto to yul.
+	// Constant is a signed 32-bit integer
+	std::ostream &operator<<(std::ostream &os, const Const &x)
+	{
+		return os << "(" << x.val() << ")";
+	}
+
+	std::ostream &operator<<(std::ostream &os, const VarRef &x)
+	{
+		return os << "a[" << (static_cast<uint32_t>(x.varnum()) % 100) << "]";
+	}
+
+	std::ostream &operator<<(std::ostream &os, const Lvalue &x)
+	{
+		return os << x.varref();
+	}
+
+	std::ostream &operator<<(std::ostream &os, const Rvalue &x)
+	{
+		if (x.has_varref())
+			return os << x.varref();
+		if (x.has_cons())
+			return os << x.cons();
+		if (x.has_binop())
+			return os << x.binop();
+		return os << "1";
+	}
+
+	std::ostream &operator<<(std::ostream &os, const BinaryOp &x)
+	{
+		switch (x.op()) {
+			case BinaryOp::ADD:
+				os << "add(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::SUB:
+				os << "sub(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::MUL:
+				os << "mul(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::DIV:
+				os << "div(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::MOD:
+				os << "mod(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::XOR:
+				os << "xor(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::AND:
+				os << "and(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::OR:
+				os << "or(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::EQ:
+				os << "eq(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::LT:
+				os << "lt(" << x.left() << "," << x.right() << ")";
+				break;
+			case BinaryOp::GT:
+				os << "gt(" << x.left() << "," << x.right() << ")";
+				break;
+		}
+		return os;
+	}
+
+	std::ostream &operator<<(std::ostream &os, const VarDecl &x)
+	{
+		os << "let x_" << x.id() << " := " << x.expr() << "\n";
```

Comment:
Just FYI: variable declarations do not necessarily need assignments - but no need to implement that for this one.

Reply:
okay